### ISSUE TITLE
Model->newInstance() eror in Laravel <7.x version

### ISSUE THREAD

**Issue by mxdfulder** [matchory/elasticsearch#8]
Model->newInstance() uses _$model->mergeCasts($this->casts);_ method from Illuminate\Database\Eloquent\Concerns\HasAttributes trait. But this method doesn't exist in that trait for Laravel version 5.x and 6.x (https://github.com/laravel/framework/blob/5.7/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php)
So any query request fails with allocating memory error.

**Comment by Radiergummi** [comment]
We currently don't really have the bandwidth right now to support and test deprecated Laravel versions. I'm happy to merge any PRs fixing this behaviour though!  
The current model implementation heavily borrows from Eloquent; I researched a little, and from what I can gather, it should be enough to add the method to the Elasticsearch model:
```php

    /**
     * Merge new casts with existing casts on the model.
     *
     * @param  array  $casts
     * @return $this
     */
    public function mergeCasts($casts)
    {
        $this->casts = array_merge($this->casts, $casts);

        return $this;
    }
```

This should override the trait method if it exists, or serve as a stand-in if it doesn't.

Would you be able to provide a PR with a test?

**Comment by mxdfulder** [comment]
Yes, I'll make PR after full test with Laravel 5.7 verstion, may be I could find some extra problems.  
And I've alredy tested your suggestion with query search, and it fixes the problem.

**Comment by mxdfulder** [comment]
Done: https://github.com/matchory/elasticsearch/pull/9 
But it may be a lot of compatibility problems with new features in old Laravel versions, and it's better replace all Database\Eloquent Concerns funcionality to avoid this.

**Comment by Radiergummi** [comment]
Hey @mxdfulder I just released `2.3.0-beta.3`, which includes all Eloquent traits as you suggested. Could you try using that release and let me know if things are working for you?

**Comment by Radiergummi** [comment]
I'll consider this done for now. If there's anything lacking, please simply open a new issue :blush:
